### PR TITLE
Fix MAPE check_metric returning NaN, enable test

### PR DESCRIPTION
**Pull request by Yard1**
Fix `check_metric` for MAPE returning NaN.

Fixes https://github.com/pycaret/pycaret/issues/458

### CODE REVIEW

**Review by Sherry40931 — COMMENTED** [review]
Is MAPE implementation wrong? It's MPE (Mean Percentage Error), not MAPE (Mean Absolute Percentage Error)

It should be:
```python
result = (np.fabs((actual.iloc[:,0] - prediction.iloc[:,0])/actual.iloc[:,0]))[mask].mean()
```
Your commit:
```python
result = (np.fabs(actual.iloc[:,0] - prediction.iloc[:,0])/actual.iloc[:,0])[mask].mean()
```

**Comment by Yard1** [comment]
The logic here has been taken from sklearn, aside from masking 0 values.

**Comment by Sherry40931** [comment]
The code of mape from sklearn:
```python
epsilon = np.finfo(np.float64).eps
mape = np.abs(y_pred - y_true) / np.maximum(np.abs(y_true), epsilon)
output_errors = np.average(mape, weights=sample_weight, axis=0)
```

[source code](https://github.com/scikit-learn/scikit-learn/blob/e21319fee/sklearn/metrics/_regression.py#L197)
[document](https://scikit-learn.org/dev/modules/generated/sklearn.metrics.mean_absolute_percentage_error.html)

**Comment by Yard1** [comment]
Interesting, thanks. I must have misremembered. I'll update it.

**Comment by Yard1** [comment]
@Sherry40931 This method of calculating MAPE has been depreciated by a new one introduced in PyCaret 2.2, therefore there's nothing to change. The new method does indeed uses the same logic as sklearn - I just confused the two.

The code for the new method used since 2.2:
```python
        def mean_absolute_percentage_error(
            y_true, y_pred, sample_weight=None, multioutput="uniform_average"
        ):
            y_type, y_true, y_pred, multioutput = _check_reg_targets(
                y_true, y_pred, multioutput
            )
            check_consistent_length(y_true, y_pred, sample_weight)
            mask = y_true != 0
            y_true = y_true[mask]
            y_pred = y_pred[mask]
            mape = np.abs(y_pred - y_true) / np.abs(y_true)
            output_errors = np.average(mape, weights=sample_weight, axis=0)
            if isinstance(multioutput, str):
                if multioutput == "raw_values":
                    return output_errors
                elif multioutput == "uniform_average":
                    # pass None as weights to np.average: uniform mean
                    multioutput = None

            return np.average(output_errors, weights=multioutput)
```

**Comment by Sherry40931** [comment]
I see. I am using 2.1.2 so I got different MAPE, I will update it.
